src/amd/compiler/aco_ir.h

```diff
@@ -719,10 +721,12 @@ struct MUBUF_instruction : public Instruction {
 struct MTBUF_instruction : public Instruction {
    unsigned dfmt; /* Data Format of data in memory buffer */
    unsigned nfmt; /* Numeric format of data in memory */
+   unsigned gfx10_img_format; /* Buffer or image format as used by GFX10 */
```

Comment:
I'd suggest somthing like:
```
union {
   struct {
      uint16_t dfmt;
      uint16_t nfmt;
   };
   uint32_t img_format;
};
```

Reply:
I don't think it matters that much. The current structure allows the code to set dfmt/nfmt and gfx10_img_format separately, which reduces the amount if `if (chip_class >= GFX10)` lines. Are you worried about the 4 bytes of extra memory? Or the extra GFX10-specific computation of this field?

Reply:
ah ok, we'll see then.

Reply:
I'm not happy with how I hacked the GFX10 formats in there right now, so maybe I'll still change this later. But for now I'd suggest to just get it in as-is, unless you have objections.

Reply:
Changed it according to your suggestion, but using a bitfield instead.